buildsafe/v1/project.proto

```diff
@@ -0,0 +1,112 @@
+// (-- api-linter: core::0191::java-package=disabled
+// (-- api-linter: core::0191::java-outer-classname=disabled
+// (-- api-linter: core::0191::java-multiple-files=disabled
+//     aip.dev/not-precedent: We're doing This as this is a Go server --)
+// (-- api-linter: core::0191::proto-package=disabled
+
+syntax = "proto3";
+
+package buildsafe.v1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+
+option go_package = "github.com/buildsafedev/bsf-apis;bsfv1";
+
+service ProjectService {
+  rpc CreateProject(CreateProjectRequest) returns (CreateProjectResponse) {
+    option (google.api.http) = {
+      post: "/v1/project"
+      body: "*"
+    };
+  }
+
+  rpc GetProject(GetProjectRequest) returns (GetProjectResponse) {
+    option (google.api.http) = {
+      get: "/v1/project/{project_id=*}"
+    };
+  }
+
+  rpc ListProjects(ListProjectsRequest) returns (ListProjectsResponse) {
+    option (google.api.http) = {
+      get: "/v1/projects/{user_id=*}"
+    };
+  }
+
+  rpc UpdateProject(UpdateProjectRequest) returns (UpdateProjectResponse) {
```

Comment:
Instead of `UpdateProject`, let's have a `RenameProject` as that is the only relevant update. 
In future, we'll have specific methods like that instead of a generic "Update" method- truly utilising the benefits of RPC.